src/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.sol

```diff
@@ -0,0 +1,158 @@
+import {IRedeemingBondingCurveFundingManagerBase} from
+    "src/modules/fundingManager/bondingCurveFundingManager/IRedeemingBondingCurveFundingManagerBase.sol";
+import {IVirtualTokenSupply} from
+    "src/modules/fundingManager/bondingCurveFundingManager/marketMaker/IVirtualTokenSupply.sol";
+import {IVirtualCollateralSupply} from
+    "src/modules/fundingManager/bondingCurveFundingManager/marketMaker/IVirtualCollateralSupply.sol";
+import {IOrchestrator} from "src/orchestrator/IOrchestrator.sol";
+
+// External Interfaces
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {IERC20MetadataUpgradeable} from
+    "@oz-up/token/ERC20/extensions/IERC20MetadataUpgradeable.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {Strings} from "@oz/utils/Strings.sol";
+
+contract BancorVirtualSupplyBondingCurveFundingManager is
+    VirtualTokenSupplyBase,
+    VirtualCollateralSupplyBase,
+    RedeemingBondingCurveFundingManagerBase
+{
+    //--------------------------------------------------------------------------
+    // Storage
+    IBancorFormula formula;
+
+    //--------------------------------------------------------------------------
+    // // Init Function
+
+    /// @inheritdoc Module
+    function init(
+        IOrchestrator orchestrator_,
+        Metadata memory metadata,
+        bytes memory configData
+    ) external override(Module) initializer {
+        __Module_init(orchestrator_, metadata);
+
+        (
+            bytes32 _name,
+            bytes32 _symbol,
+            uint8 _decimals,
+            address _formula,
+            uint _initalTokenSupply,
+            uint _initialCollateralSupply
+        ) = abi.decode(
+            configData, (bytes32, bytes32, uint8, address, uint, uint)
+        );
+
+        __ERC20_init(
+            string(abi.encodePacked(_name)), string(abi.encodePacked(_symbol))
+        );
+
+        formula = IBancorFormula(_formula);
+
+        _setTokenDecimals(_decimals);
+        _setVirtualCollateralSupply(_initialCollateralSupply);
+        _setVirtualTokenSupply(_initalTokenSupply);
+        _setCollateral(address(__Module_orchestrator.token()));
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Functions
+
+    function buyOrder(uint _depositAmount)
+        external
+        payable
+        override(BondingCurveFundingManagerBase)
+    {
+        // WiP
+        // Deduct fee from incoming value. Fee is paid in collateral token
+        uint amountIssued = _issueTokens(_depositAmount, collateral);
+        _addTokenAmount(amountIssued);
+        _addCollateralAmount(_depositAmount);
+    }
+
+    function sellOrder(uint _depositAmount)
+        external
+        payable
+        override(RedeemingBondingCurveFundingManagerBase)
+    {
+        // WiP
+        // Q: Deduct fee from token or collateral?
```

Comment:
When selling, deducting the fee from the collateral after the token has been burned simplifies the process: you use the math as intended and end up with collected fees in only one token.